specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2025-07-01-preview/PreviewAlertRule.json

```diff
@@ -0,0 +1,510 @@
+            "modelAsString": true
+          },
+          "description": "The state of the alert after the evaluation."
+        },
+        "evaluatedPeriods": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/EvaluatedPeriod"
+          },
+          "description": "An array of evaluated periods."
+        },
+        "thresholdMet": {
+          "type": "boolean",
+          "description": "Indicates whether the threshold was met for this evaluation."
+        }
+      },
+      "description": "An evaluation of the alert rule."
+    },
+    "EvaluatedPeriod": {
+      "type": "object",
+      "properties": {
+        "timestamp": {
+          "type": "string",
+          "format": "date-time",
+          "description": "The timestamp of the evaluated period."
+        },
+        "highThreshold": {
+          "type": "number",
+          "format": "double",
+          "description": "The high threshold value for the evaluation period."
+        },
+        "lowThreshold": {
+          "type": "number",
+          "format": "double",
+          "description": "The low threshold value for the evaluation period."
+        },
+        "metricValue": {
+          "type": "number",
+          "format": "double",
+          "description": "The metric value for the evaluation period."
+        },
+        "thresholdMet": {
+          "type": "boolean",
+          "description": "Indicates whether the threshold was met during the evaluation period."
+        }
+      },
+      "description": "Details of the evaluated period."
+    },
+    "DimensionNameAndValue": {
+      "type": "object",
+      "description": "Dimension name and value",
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "Name of the dimension"
+        },
+        "value": {
+          "type": "string",
+          "description": "Value of the dimension"
+        }
+      },
+      "required": [
+        "name",
+        "value"
+      ]
+    },
+    "PreviewAlertRuleRequest": {
+      "type": "object",
+      "properties": {
+        "location": {
+          "type": "string",
+          "description": "The location of the rule resource."
+        },
+        "properties": {
+          "type": "object",
+          "properties": {
+            "timespan": {
+              "type": "string",
+              "format": "duration",
+              "description": "Specifies the timespan of the preview in ISO 8601 duration format."
+            },
+            "scheduledQueryRuleProperties": {
+              "$ref": "#/definitions/LogAlertRuleResource",
+              "description": "The properties of the alert rule to preview."
+            }
+          },
+          "required": [
+            "timespan"
+          ]
+        }
+      },
+      "required": [
+        "location",
+        "properties"
+      ]
+    }
+  },
+  "parameters": {
+    "ResourceUriParameter": {
+      "name": "resourceUri",
```

Comment:
>resourceUri [](http://example.com/codeflow?start=15&length=11)

This is actually a resourceId, not resourceUri.
This: "/providers/Microsoft.Blorp/blurps/myBlurp",
not this: "https:://management.azure.com/providers/MIcrosoft.Blorp/blurps/myBlurp".

The name is probably not really used for anything, but resourceId would be correct.